cordova-common/src/CordovaLogger.js

```diff
@@ -153,6 +153,30 @@ CordovaLogger.prototype.setLevel = function (logLevel) {
 };
 
 /**
+ * Adjusts the current logger level according to the passed options.
+ *
+ * @param   {Object|Array}  opts  An object or args array with options
+ *
+ * @return  {CordovaLogger}     Current instance, to allow calls chaining.
+ */
+CordovaLogger.prototype.adjustLoggerLevel = function (opts) {
+    if (opts instanceof Array) {
+        opts.silent = opts.indexOf('--silent') !== -1;
+        opts.verbose = opts.indexOf('--verbose') !== -1 || opts.indexOf('-d') !== -1;
+    }
+
+    if (opts.silent) {
+        this.setLevel('error');
+    }
+
+    if (opts.verbose || opts.d) {

```

Comment:
I think that `opts.d` is not needed because nopt converts `-d` to `{verbose: true}` if `-d` is a shortcut for `-verbose`